### ISSUE TITLE
Incomplete installation of Deskflow from release artifacts

### ISSUE THREAD

**Issue by AnrDaemon** [deskflow/deskflow#7763]
### Project

Deskflow

### Deskflow version number

1.17.0.87

### Build type

Deskflow package (downloaded from Deskflow)

### Operating systems (OS)

- [X] Windows
- [ ] macOS
- [ ] Linux (X11)
- [ ] Linux (Wayland)
- [ ] BSD-derived
- [ ] Other (please specify)

### Wayland on Linux

- [ ] I have reviewed the Wayland known issues and my issue is new
- [ ] I am not using Wayland on Linux

### Signing on macOS

- [ ] I have authorized the app to run on my Mac
- [ ] I am not using macOS

### Continuous build

- [X] I have tried the latest continuous build and the issue persists
- [ ] I am unable to try the latest continuous build

### OS versions/distros

Microsoft Windows [Version 10.0.19045.5011]

### Deskflow configuration

Stock install.

### What steps will reproduce the problem?

1. Install deskflow.
2. Run deskflow.exe.
3. Popup "Unable to find `brotlidec.dll`."

### Log output

```shell
None.
```

**Comment by AnrDaemon** [comment]
+ zstd.dll

**Comment by AnrDaemon** [comment]
+ libpng16

**Comment by AnrDaemon** [comment]
Sorry, there's just too many libpng16's, can't investigate further.

**Comment by AnrDaemon** [comment]
Though, here:

```
$ cygcheck ./deskflow.exe > /dev/null
cygcheck: track_down: could not find icuin74.dll

cygcheck: track_down: could not find icuuc74.dll

cygcheck: track_down: could not find double-conversion.dll

cygcheck: track_down: could not find pcre2-16.dll

cygcheck: track_down: could not find zstd.dll

cygcheck: track_down: could not find libpng16.dll

cygcheck: track_down: could not find harfbuzz.dll

cygcheck: track_down: could not find freetype.dll

cygcheck: track_down: could not find brotlidec.dll

cygcheck: track_down: could not find zstd.dll
```

I hope this helps a little bit.

**Comment by sithlord48** [comment]
can you see if this fixes the first one , at least ill know im on the right track
[Hmm thought ](https://github.com/deskflow/deskflow/actions/runs/11419285142/artifacts/2077836006)

**Comment by AnrDaemon** [comment]
There's only test results .md file in the archive.

**Comment by sithlord48** [comment]
oh sorry this one is the zip w/ windows packages https://github.com/deskflow/deskflow/actions/runs/11419285142/artifacts/2077836557

**Comment by AnrDaemon** [comment]
Nop, still the same error pops up on starting deskflow.exe
Same report from cygcheck.
Attempt to start -client or -server app ends with
```
[2024-10-21T09:54:41] FATAL: a runtime error occurred: This should be caught in your main function, see examples
```

**Comment by nbolton** [comment]
Thanks for raising, I'll investigate this now.

**Comment by nbolton** [comment]
I can confirm there is an issue with opening `deskflow.exe`:
![image](https://github.com/user-attachments/assets/1e973f17-845f-4ac7-b5c8-68e9faba4a51)

I will bisect to find the commit that caused this, it's probably very recent.

**Comment by nbolton** [comment]
Bisect done.

- Bad: bf4a762124a2de8bb298ff0629267223745e6ef1
- Good: 7199e5f1704ab9b879128c43b554509da702d4e3

**Comment by sithlord48** [comment]
I suspect the installer script was doing something a bit odd. All these libs should be deployed when windeployqt is called.

**Comment by nbolton** [comment]
Thanks for fix, @sithlord48! #7783

For those interested, we forgot to bundle these DLLs in the WiX [`Product.wxs`](https://github.com/deskflow/deskflow/blob/2136af68b542fa1ec185a56c03535f03bb587170/res/dist/wix/Product.wxs) file:
```
<File Source="$(var.BinDir)\brotlicommon.dll"/>
<File Source="$(var.BinDir)\brotlidec.dll"/>
<File Source="$(var.BinDir)\bz2.dll"/>
<File Source="$(var.BinDir)\double-conversion.dll"/>
<File Source="$(var.BinDir)\freetype.dll"/>
<File Source="$(var.BinDir)\harfbuzz.dll"/>
<File Source="$(var.BinDir)\icudt74.dll"/>
<File Source="$(var.BinDir)\icuin74.dll"/>
<File Source="$(var.BinDir)\icuuc74.dll"/>
<File Source="$(var.BinDir)\libpng16.dll"/>
<File Source="$(var.BinDir)\pcre2-16.dll"/>
<File Source="$(var.BinDir)\zlib1.dll"/>
<File Source="$(var.BinDir)\zstd.dll"/>
```

Previously we were building `wintoastlib.cpp` as part of our binary (which is similar to, but worse than static linking).

Related: #7784
